app/models/concerns/hostext/operating_system.rb

```diff
@@ -0,0 +1,55 @@
+module Hostext
+  module OperatingSystem
+    extend ActiveSupport::Concern
+    included do
+      scope :with_os, -> { where('hosts.operatingsystem_id IS NOT NULL') }
+      alias_attribute :os, :operatingsystem
+      validates :operatingsystem_id, :presence => true,
+        :if => ->(host) { host.managed }
+    end
+
+    # returns a configuration template (such as kickstart) to a given host
+    def provisioning_template(opts = {})
+      opts[:kind]               ||= "provision"
+      opts[:operatingsystem_id] ||= operatingsystem_id
+      opts[:hostgroup_id]       ||= hostgroup_id
+      opts[:environment_id]     ||= environment_id
+
+      ProvisioningTemplate.find_template opts
+    end
+
+    def available_template_kinds(provisioning = nil)
```

Comment:
Do you think, we can clean that one a bit up? Maybe extracting this to two methods? The if looks very scary.